__tests__/integration/utils/test-utils-integration.js

```diff
@@ -794,18 +794,42 @@ export class TestUtils {
   static async waitNewBlock(currentHeight = null) {
     let baseHeight = currentHeight;
 
+    // This timeout is a protection, so the integration tests
+    // don't keep running in case of a problem
+    // After using the timeout as 120s, we had some timeouts
+    // because the CI runs in a free github runner
+    // so we decided to increase this timeout to 600s, so
+    // we don't have this error anymore
+    const timeout = 600000;
+    let timeoutReached = false;
+    // Timeout handler
+    const timeoutHandler = setTimeout(() => {
+      timeoutReached = true;
+    }, timeout);
+
     if (!baseHeight) {
       baseHeight = await TestUtils.getFullNodeNetworkHeight();
     }
 
     let networkHeight = baseHeight;
 
     while (networkHeight === baseHeight) {
+      if (timeoutReached) {
+        break;
+      }
+
       networkHeight = await TestUtils.getFullNodeNetworkHeight();
 
       await delay(1000);
     }
 
+    // Clear timeout handler
```

Comment:
It is clear enough what the next line is doing, I don't think we need this comment